src/js/components/SchemaForm.js

```diff
@@ -53,6 +53,13 @@ class SchemaForm extends React.Component {
     this.setState({renderGemini: true});
   }
 
+  componentWillUnmount() {
+    // Unschedule all validation if component unmounts.
+    if (this.timer) {
+      clearTimeout(this.timer);
```

Comment:
after all of the `clearTimeout`s, do we want to do `this.timer = null` or something?